### PR TITLE
Spam-reduction

### PR DESCRIPTION
**Pull request by CallumWalley**
Changes:

* Don't prompt for RH telemetry.
* Don't prompt for 'create new branch on remote'.
* Removed GH actions plugin. (not needed for contribution, saves being prompted to login).
* Prompt user for commit message in dropbox, rather than opening editor (possibly controversial).
* Added gitlens (possibly controversial, does not reduce spam).
* Added back `severity: "info"`. (GitHub actions problem matcher calls these `notice`s, meaning the pattern is not matched, so fallback `info` will be used, as is correct.)

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
                 "GitHub.vscode-github-actions",
                 "GitHub.vscode-pull-request-github",
                 "jebbs.markdown-extended",
-                "Gruntfuggly.triggertaskonsave"
+                "Gruntfuggly.triggertaskonsave",
+                "eamodio.gitlens"
```
Comment (jennan) [inline .devcontainer/devcontainer.json:30]: I would prefer to not add gitlens for now. It adds a lot of features and buttons... which I fear would be distracting for a new less technical users. I prefer to keep it simpler for codespace.

File: .vscode/default.code-profile
Comment (jennan) [inline .vscode/default.code-profile:1]: Can you explain in few words what is this doing :-)?

File: .vscode/tasks.json [hunk 2]
```diff
@@ -17,7 +17,9 @@
                 "showReuseMessage": false
             },
             "command": "./checks/run_meta_check.py",
+
             "problemMatcher": {
+                "severity": "info",
```
Comment (jennan) [inline .vscode/tasks.json:22]: As there is a severity field in the matcher, I don't see why the default severity is needed. Please can you elaborate on this :)?

**Review by jennan — CHANGES_REQUESTED** [review]
Hi Callum, sorry I have some questions before approving and I'd like gitlens to not be in this PR for codespace if possible (we can discuss this of course).

File: .vscode/tasks.json (hunk 2 above)
Comment (CallumWalley) [reply .vscode/tasks.json:22]: GitHub 'annotations' (the actions problem matcher) can be one of three severities, '`ERROR` `WARNING` or ***`NOTICE`***. All of the checks outputs are made first to conform to this schema. VSCode problem matchers also recognise three levels, `ERROR` `WARNING` and ***`INFO`***. This means any `NOTICE` level matches have their severity level ignored, and will use the default severity `ERROR`. Since everything else should have a severity level, only notices map to info and everything is as it should be.

TDLR; When this line is removed, all `INFO` level messages become `ERROR`'s

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (CallumWalley) [reply .devcontainer/devcontainer.json:30]: I agree, GitLens is a bit overwhelming.
(Although I received some positive feedback a co-worker after the demo about the inline git blame messages).

My preference would be for keeping it, but disabling/hiding all the unused bits.

File: .vscode/default.code-profile
Comment (CallumWalley) [reply .vscode/default.code-profile:1]: Things that aren't 'settings' (like layout, what's in the action bar, and some "don't ask again" flags) are stored as 'Profile' data. 

Profiles can be exported and imported to file. (Unfortunately it is this weird JSON-ish format where every object is escaped, which is why you see so many backslashes in the more nested part).

I added this file to show, 

1. This works and is a viable way of hiding some of the distractions and popups. (Click on the cog icon bottom left, then 'profiles' -> 'import profiles' -> 'from file' - `/workspaces/support-docs-concept/.vscode/default.code-profile`. Will reload container, but you will have the desired environment.)
2. You can manually edit the `.code-profile` file, and only include the bits you want changed.

Comment (CallumWalley) [reply .vscode/default.code-profile:1]: Sorry, that was lots of words.

TLDR; Demonstrates one way we could remove visual clutter from the environment, by hiding some icons, and pre-ticking "Don't ask me this again"

File: .vscode/tasks.json (hunk 2 above)
Comment (jennan) [reply .vscode/tasks.json:22]: Ah ok, it explains some things I have seen ERRORS when it should have been notices. Thank you for figuring this out :-).

File: .vscode/default.code-profile
Comment (jennan) [reply .vscode/default.code-profile:1]: That looks great. I would still prefer to have this automated (no need for user to do any manipulation) before adding merging it in main, if the goal is to declutter codespace for first time users.

**Comment by jennan** [comment]
Following on a discussion with @CallumWalley , I think we agreed on
- keeping this PR opened until gitlens can be de-cluttered and profile be automatically activated in codespace
- however I am happy with all the other changes, they can be merged into main (cherry-pick or open another PR with just these changes).

**Comment by CallumWalley** [comment]
Closing PR to avoid clutter. Leaving branch open for now
